test/Microsoft.ML.Tests/OnnxEstimatorTests.cs

```diff
@@ -0,0 +1,189 @@
+            var sizeData = new List<TestDataSize> { new TestDataSize() { data_0 = new float[2] } };
+            var pipe = new OnnxEstimator(Env, modelFile, new[] { "data_0" }, new[] { "softmaxout_1" });
+
+            var invalidDataWrongNames = ComponentCreation.CreateDataView(Env, xyData);
+            var invalidDataWrongTypes = ComponentCreation.CreateDataView(Env, stringData);
+            var invalidDataWrongVectorSize = ComponentCreation.CreateDataView(Env, sizeData);
+            TestEstimatorCore(pipe, dataView, invalidInput: invalidDataWrongNames);
+            TestEstimatorCore(pipe, dataView, invalidInput: invalidDataWrongTypes);
+
+            pipe.GetOutputSchema(SchemaShape.Create(invalidDataWrongVectorSize.Schema));
+            try
+            {
+                pipe.Fit(invalidDataWrongVectorSize);
+                Assert.False(true);
+            }
+            catch (ArgumentOutOfRangeException) { }
+            catch (InvalidOperationException) { }
+        }
+
+        [Fact]
+        void TestModelLoadAndScore()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                return;
+
+            var modelFile = "squeezenet/00000001/model.onnx";
+
+            var samplevector = getSampleArrayData();
+
+            var dataView = ComponentCreation.CreateDataView(Env,
+                new List<TestData>(new TestData[] {
+                    new TestData()
+                    {
+                        data_0 = samplevector
+                    }
+                }));
+
+            var inputNames = new[] { "data_0" };
+            var outputNames = new[] { "softmaxout_1" };
+            var est = new OnnxEstimator(Env, modelFile, inputNames, outputNames);
+            var transformer = est.Fit(dataView);
+            var result = transformer.Transform(dataView);
+            var resultRoles = new RoleMappedData(result);
+            using (var ms = new MemoryStream())
+            {
+                TrainUtils.SaveModel(Env, Env.Start("saving"), ms, null, resultRoles);
+                ms.Position = 0;
+                var loadedView = ModelFileUtils.LoadTransforms(Env, dataView, ms);
+                //ValidateTensorFlowTransformer(loadedView);
+
+                loadedView.Schema.TryGetColumnIndex(outputNames[0], out int softMaxOut1);
+                using (var cursor = loadedView.GetRowCursor(col => col == softMaxOut1))
+                {
+                    VBuffer<float> softMaxValue = default;
+                    var softMaxGetter = cursor.GetGetter<VBuffer<float>>(softMaxOut1);
+                    while (cursor.MoveNext())
+                    {
+                        softMaxGetter(ref softMaxValue);
+                        Console.WriteLine(softMaxValue.Values[0]);
+                        Console.WriteLine(softMaxValue.Values[1]);
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public void OnnxStatic()
```

Comment:
Not related to this one, but I would love to have test which would ran ML.Net-> Onnx conversion and then score that model, and compare results.

Reply:
Yup, that's a great Uber-test to have. We can probably track that as a separate feature or task (since it's using 2 diff transforms). Let me know if we can close this for now.

---
In reply to: [219967750](https://github.com/dotnet/machinelearning/pull/942#discussion_r219967750) [](ancestors = 219967750)